### PR TITLE
DOC: Update doc and internal links after uxl transfer

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

After this repository got transferred to uxlfoundation, the doc links for github do not work anymore as they do not redirect to the new URL.

This PR changes all the instances of string "intel.github.io" to "uxlfoundation.github.io" which should be the new URL for these docs.

It also modifies internal links to files within the repository to point to `uxlfoundation/scikit-learn-intelex` instead.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.

**Performance**

Not applicable.

### CODE REVIEW

**Review by icfaust — APPROVED** [review]

**Comment by homksei** [comment]
Should occurrences of `oneapi-src` be replaced with `uxlfoundation` where possible in this pull request ?

**Comment by david-cortes-intel** [comment]
> Should occurrences of `oneapi-src` be replaced with `uxlfoundation` where possible in this pull request ?

Yes, thanks for pointing this out - I've replaced those too now.

File: doc/sources/samples.rst [hunk 1]
```diff
@@ -65,11 +65,11 @@ If you want to run them locally, refer to `these instructions
 .. toctree::
    :maxdepth: 1
 
-   Intel® Extension for Scikit-Learn* Getting Started Sample <https://github.com/oneapi-src/oneAPI-samples/tree/master/AI-and-Analytics/Getting-Started-Samples/Intel_Extension_For_SKLearn_GettingStarted>
-   Intel® Extension for Scikit-Learn*: SVC for Adult dataset Performance Sample <https://github.com/oneapi-src/oneAPI-samples/tree/master/AI-and-Analytics/Features-and-Functionality/Intel_Extension_For_SKLearn_Performance_SVC_Adult>
+   Intel® Extension for Scikit-Learn* Getting Started Sample <https://github.com/uxlfoundation/oneAPI-samples/tree/master/AI-and-Analytics/Getting-Started-Samples/Intel_Extension_For_SKLearn_GettingStarted>
+   Intel® Extension for Scikit-Learn*: SVC for Adult dataset Performance Sample <https://github.com/uxlfoundation/oneAPI-samples/tree/master/AI-and-Analytics/Features-and-Functionality/Intel_Extension_For_SKLearn_Performance_SVC_Adult>
```
Comment (homksei) [inline doc/sources/samples.rst:69]: `oneAPI-samples` are currently in `oneapi-src`. It makes sense to leave the links as they are, at least for now.

File: doc/sources/samples.rst [hunk 2]
```diff
@@ -65,11 +65,11 @@ If you want to run them locally, refer to `these instructions
...
 
 .. seealso::
 
    There are also :ref:`Kaggle kernels <kaggle>` that use |intelex| for a variety of machine learning scenarios.
 
-.. note:: Explore the complete list of oneAPI code samples in the `oneAPI Samples Catalog <https://oneapi-src.github.io/oneAPI-samples/>`_. These samples were designed to help you develop, offload, and optimize multiarchitecture applications targeting CPUs, GPUs, and FPGAs.
+.. note:: Explore the complete list of oneAPI code samples in the `oneAPI Samples Catalog <https://uxlfoundation.github.io/oneAPI-samples/>`_. These samples were designed to help you develop, offload, and optimize multiarchitecture applications targeting CPUs, GPUs, and FPGAs.
```
Comment (homksei) [inline doc/sources/samples.rst:75]: and here

File: doc/sources/samples.rst (hunk 1 above)
Comment (david-cortes-intel) [reply doc/sources/samples.rst]: Thanks, updated both of them.

File: doc/sources/samples.rst [hunk 3]
```diff
@@ -66,7 +66,7 @@ If you want to run them locally, refer to `these instructions
    :maxdepth: 1
 
    Intel® Extension for Scikit-Learn* Getting Started Sample <https://github.com/oneapi-src/oneAPI-samples/tree/master/AI-and-Analytics/Getting-Started-Samples/Intel_Extension_For_SKLearn_GettingStarted>
-   Intel® Extension for Scikit-Learn*: SVC for Adult dataset Performance Sample <https://github.com/oneapi-src/oneAPI-samples/tree/master/AI-and-Analytics/Features-and-Functionality/Intel_Extension_For_SKLearn_Performance_SVC_Adult>
+   Intel® Extension for Scikit-Learn*: SVC for Adult dataset Performance Sample <https://github.com/uxlfoundation/oneAPI-samples/tree/master/AI-and-Analytics/Features-and-Functionality/Intel_Extension_For_SKLearn_Performance_SVC_Adult>
```
Comment (homksei) [inline doc/sources/samples.rst:69]: Please, change it as well. I must have missed it during the first review.

**Review by homksei — APPROVED** [review]

**Review by napetrov — APPROVED** [review]

**Pull request merged**